RULES/libsupcxx/xlib/xlib.go

```diff
@@ -0,0 +1,76 @@
+package xlib
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"dbt-rules/RULES/cc"
+	"dbt-rules/RULES/core"
+
+	"libsupcxx/RULES/target"
+	"libsupcxx/libsupcxx/include"
+)
+
+// This package is only meant to be used within libsupcxx, outside code should use
+// libsupcxx/RULES/libsupcxx. See there for info about XLibrary.
+type XLibrary struct {
+	Out           core.OutPath
+	Srcs          []core.Path
+	Includes      []core.Path
+	CompilerFlags []string
+	XDeps         []XLibrary
+	Shared        bool
+	AlwaysLink    bool
+	Targets       []target.Target
+}
+
+func ContainsTarget(ts []target.Target, t target.Target) bool {
```

Comment:
`BuildsTarger` or `DefinesTarget` would be better.

Reply:
This was more meant as a .contains function for a slice of targets. I will probably move it to target. It is also useful when checking for duplicates in target.Register()

Reply:
Ok, use your judgement.